presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/EvaluateProjectOnValues.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.metadata.FunctionAndTypeManager;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.ProjectNode;
+import com.facebook.presto.spi.plan.ValuesNode;
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.Interpreters;
+import com.facebook.presto.sql.planner.RowExpressionInterpreter;
+import com.facebook.presto.sql.planner.VariableResolver;
+import com.facebook.presto.sql.planner.iterative.Rule;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.SystemSessionProperties.isEvaluateProjectOnValues;
+import static com.facebook.presto.spi.relation.ExpressionOptimizer.Level.EVALUATED;
+import static com.facebook.presto.spi.relation.ExpressionOptimizer.Level.OPTIMIZED;
+import static com.facebook.presto.sql.planner.plan.Patterns.project;
+import static com.facebook.presto.sql.planner.plan.Patterns.source;
+import static com.facebook.presto.sql.planner.plan.Patterns.values;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * This optimizer looks for ProjectNode followed by a ValuesNode and get the ProjectNode Evaluated.
+ * <p/>
+ * Plan before optimizer:
+ * <pre>
+ * ProjectNode (outputVariables)
+ *   - ValuesNode
+ * </pre>
+ * <p/>
+ * Plan after optimizer:
+ * <pre>
+ * ValuesNode (outputVariables)
+ * </pre>
+ */
+public class EvaluateProjectOnValues
+        implements Rule<ProjectNode>
+{
+    private static final Pattern<ProjectNode> PATTERN = project().with(source().matching(values()));
+
+    private final FunctionAndTypeManager functionAndTypeManager;
+
+    public EvaluateProjectOnValues(FunctionAndTypeManager functionAndTypeManager)
+    {
+        requireNonNull(functionAndTypeManager, "functionManager is null");
+        this.functionAndTypeManager = functionAndTypeManager;
```

Comment:
nit: You can combine this into a single line
```suggestion
        this.functionAndTypeManager = requireNonNull(functionAndTypeManager, "functionManager is null");
```

Reply:
fixed